go_sequencer/database/statedb/score_service.go

```diff
@@ -0,0 +1,142 @@
+package statedb
+
+import (
+	"errors"
+	"tokamak-sybil-resistance/common"
+
+	"github.com/iden3/go-merkletree"
+	"github.com/iden3/go-merkletree/db"
+)
+
+var (
+	// ErrScoreAlreadyExists is used when CreateScore is called and the
+	// Score already exists
+	ErrScoreAlreadyExists = errors.New("cannot CreateScore becase Score already exists")
+	// PrefixKeyScoIdx is the key prefix for accountIdx in ScoreTree
+	PrefixKeyScoIdx = []byte("s:")
+)
+
+// CreateScore creates a new Score in the StateDB for the given Idx. If
+// StateDB.MT==nil, MerkleTree is not affected, otherwise updates the
+// MerkleTree, returning a CircomProcessorProof.
+func (s *StateDB) CreateScore(idx common.AccountIdx, score *common.Score) (
+	*merkletree.CircomProcessorProof, error) {
+	cpp, err := CreateScoreInTreeDB(s.db.DB(), s.ScoreTree, idx, score)
+	if err != nil {
+		return cpp, common.Wrap(err)
+	}
+	return cpp, nil
+}
+
+// CreateScoreInTreeDB is abstracted from StateDB to be used from StateDB and
+// from ExitTree. Creates a new Score in the StateDB for the given Idx. If
+// StateDB.MT==nil, MerkleTree is no affected, otherwise updates the
+// MerkleTree, returning a CircomProcessorProof
+func CreateScoreInTreeDB(sto db.Storage, mt *merkletree.MerkleTree, idx common.AccountIdx,
```

Comment:
In the create and update function here - 

The general transactions part is repeated and could be moved to a different function. Similarly we can do for vouches too Ref - [Link](https://github.com/tokamak-network/tokamak-sybil-resistance/pull/31/files#diff-badd22c6f5a1920fdddb926446fdc32fa533915c33c26a0ad81e1a6dd9cd752d)